[Swift in WebKit] The Swift implementation of WKMarketplaceKit is incorrect

rr-codes opened the pull request:
#### 238e4dd47de1e21bbc88ba1575ed1ef4a0f034f0
<pre>
[Swift in WebKit] The Swift implementation of WKMarketplaceKit is incorrect
<a href="https://bugs.webkit.org/show_bug.cgi?id=292580">https://bugs.webkit.org/show_bug.cgi?id=292580</a>
<a href="https://rdar.apple.com/150729230">rdar://150729230</a>

Reviewed by Elliott Williams.

Similar to 294442@main, these types were also incorrectly implemented in Swift.
Fix by using the built-in standard language feature instead.

* Source/WebKit/Configurations/WebKitSwift.xcconfig:

There is no need to exclude files anymore now that we can just use normal compile time conditionals.

* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:

- Remove the guard around WebKitSwiftSoftLink.h for consistency with how other import
of this file work.
- Move the soft linking macros from this semi-random place to the centeralized `WebKitSwiftSoftLink.h` file.
- Remove the forward declaration since it is now unneeded since the header itself can just be imported as normal.
- Add `USE(APPLE_INTERNAL_SDK)` to the existing conditionals since this was implicitly
a requirement anyways, due to the fact that the Swift files were being excluded anyways
when the internal SDK was not being used.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.h: Copied from Source/WebKit/WebKitSwift/WebKitSwift.h.
* Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.swift: Renamed from Source/WebKit/WebKitSwift/MarketplaceKit/MarketplaceKitWrapper.swift.

- Move the header to WKS for consistency
- Rename everything `WKMarketplaceKit` for consistency

(WKMarketplaceKit.requestAppInstallation(withTopOrigin:url:)):

- Use `internal import` where applicable
- Use the proper facilities for the Swift impl of an Obj C header
- Use modern `async throws`, which greatly simplifies the code

* Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift:
* Source/WebKit/WebKitSwift/WebKitSwift.h:

Canonical link: <a href="https://commits.webkit.org/294560@main">https://commits.webkit.org/294560@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/464457746f5ad75c103c1537cf3609b49980e0a9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102245 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21913 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12228 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107404 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52881 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22221 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30420 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77798 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34787 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105251 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17196 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92298 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58135 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17027 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52239 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10396 "Exiting early after 60 failures. 76911 tests run. 60 failures") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109780 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29377 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21654 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86780 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29740 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88501 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86368 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21977 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31174 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8893 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/23619 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29305 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34600 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29116 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32439 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30675 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.swift:
```diff
@@ -21,32 +21,27 @@
 // THE POSSIBILITY OF SUCH DAMAGE.
 
-#if os(iOS) && !os(visionOS) && canImport(MarketplaceKit)
+#if USE_APPLE_INTERNAL_SDK && HAVE_MARKETPLACE_KIT
```
Pushing USE_APPLE_INTERNAL_SDK checks down into the code makes sense, though I wonder if it _should_ be this way. MarketplaceKit is a public framework; maybe we can make this work on the open source build without much effort?

rr-codes replied:
yes, that was my plan for my next PR! just wanted to not make too many different changes in a single PR

rr-codes committed:
[Swift in WebKit] The Swift implementation of WKMarketplaceKit is incorrect
https://bugs.webkit.org/show_bug.cgi?id=292580
rdar://150729230

Reviewed by Elliott Williams.

Similar to 294442@main, these types were also incorrectly implemented in Swift.
Fix by using the built-in standard language feature instead.

* Source/WebKit/Configurations/WebKitSwift.xcconfig:

There is no need to exclude files anymore now that we can just use normal compile time conditionals.

* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.h:
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:

- Remove the guard around WebKitSwiftSoftLink.h for consistency with how other import
of this file work.
- Move the soft linking macros from this semi-random place to the centeralized `WebKitSwiftSoftLink.h` file.
- Remove the forward declaration since it is now unneeded since the header itself can just be imported as normal.
- Add `USE(APPLE_INTERNAL_SDK)` to the existing conditionals since this was implicitly
a requirement anyways, due to the fact that the Swift files were being excluded anyways
when the internal SDK was not being used.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.h: Copied from Source/WebKit/WebKitSwift/WebKitSwift.h.
* Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.swift: Renamed from Source/WebKit/WebKitSwift/MarketplaceKit/MarketplaceKitWrapper.swift.

- Move the header to WKS for consistency
- Rename everything `WKMarketplaceKit` for consistency

(WKMarketplaceKit.requestAppInstallation(withTopOrigin:url:)):

- Use `internal import` where applicable
- Use the proper facilities for the Swift impl of an Obj C header
- Use modern `async throws`, which greatly simplifies the code

* Source/WebKit/WebKitSwift/TextAnimation/TextAnimationManager.swift:
* Source/WebKit/WebKitSwift/WebKitSwift.h:

Canonical link: https://commits.webkit.org/294560@main
